src/network/clientpackethandler.cpp

```diff
@@ -215,11 +215,28 @@ void Client::handleCommand_AccessDenied(NetworkPacket* pkt)
 
 		u8 denyCode = SERVER_ACCESSDENIED_UNEXPECTED_DATA;
 		*pkt >> denyCode;
-		if (denyCode == SERVER_ACCESSDENIED_CUSTOM_STRING) {
+		if (denyCode == SERVER_ACCESSDENIED_SHUTDOWN ||
+				denyCode == SERVER_ACCESSDENIED_CRASH) {
 			*pkt >> m_access_denied_reason;
-		}
-		else if (denyCode < SERVER_ACCESSDENIED_MAX) {
+			if (m_access_denied_reason == "") {
+				m_access_denied_reason = accessDeniedStrings[denyCode];
+			}
+			u8 reconnect;
+			*pkt >> reconnect;
+			m_access_denied_reconnect = reconnect & 1;
```

Comment:
You realize the & 1 isn't necessary, right?  Check out the C++ spec on conversions to boolean.